opentelemetry/src/attributes/set.rs

```diff
@@ -0,0 +1,244 @@
+use once_cell::sync::Lazy;
+use std::collections::hash_map::DefaultHasher;
+use std::collections::HashSet;
+use std::sync::Arc;
+use std::{
+    cmp::Ordering,
+    hash::{Hash, Hasher},
+};
+
+use crate::{Array, Key, KeyValue, Value};
+use ordered_float::OrderedFloat;
+
+#[derive(Clone, Debug)]
+struct HashKeyValue(KeyValue);
+
+impl Hash for HashKeyValue {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.0.key.hash(state);
+        match &self.0.value {
+            Value::F64(f) => OrderedFloat(*f).hash(state),
+            Value::Array(a) => match a {
+                Array::Bool(b) => b.hash(state),
+                Array::I64(i) => i.hash(state),
+                Array::F64(f) => f.iter().for_each(|f| OrderedFloat(*f).hash(state)),
+                Array::String(s) => s.hash(state),
+            },
+            Value::Bool(b) => b.hash(state),
+            Value::I64(i) => i.hash(state),
+            Value::String(s) => s.hash(state),
+        };
+    }
+}
+
+impl PartialOrd for HashKeyValue {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for HashKeyValue {
+    fn cmp(&self, other: &Self) -> Ordering {
+        match self.0.key.cmp(&other.0.key) {
```

Comment:
I understand this is not introduced in this PR, but I am not able to follow the logic behind this comparison.. Wont a simple "self.0.key.cmp(&other.0.key)" be sufficient here? The question of Key's being equal should not occur as we de-dedup before sorting...

Reply:
@KallDrexx Gentle reminder for this! Not a blocker for this PR.